src/frontend/js/components/AshleyEditor/LatexButton/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useRef } from 'react';
+import { DraftJsButtonTheme } from '@draft-js-plugins/buttons';
+import { EditorState } from 'draft-js';
+import {
+  insertInlineTeX,
+  insertTeXBlock,
+} from 'draft-js-latex-plugin/lib/utils/insertTeX';
+import { TypeLatexStyle } from '../../../types/Enums';
+interface LatexButtonProps {
+  type: string;
+  theme: DraftJsButtonTheme;
+  getEditorState: () => EditorState;
+  setEditorState: (editorState: EditorState) => void;
+}
+
+export const LatexButton = (props: LatexButtonProps) => {
+  const node = useRef<HTMLDivElement>(null);
+  const { buttonWrapper = '', button } = props.theme;
+  const imageClassName =
+    props.type === TypeLatexStyle.INLINE ? 'latex-inline' : 'latex-block';
+  const handleAddLatex = (
+    e: React.MouseEvent<HTMLButtonElement>,
+    type: string,
+  ) => {
+    e.preventDefault();
+    let newEditorState;
+    const editorState = props.getEditorState();
+    if (type === TypeLatexStyle.BLOCK) {
+      newEditorState = insertTeXBlock(editorState);
+    } else {
+      newEditorState = insertInlineTeX(editorState);
+    }
+    props.setEditorState(newEditorState);
+    return 'handled';
```

Comment:
Is `"handled"` something waited by draft.js to know to know which type of action has been triggered ? If yes, what do you think to create a enum to store those action names ?